feat(replays): Add documentation for `beforeAddRecordingEvent` (#7103)
getsentry/sentry-docs: src/platforms/javascript/common/session-replay/privacy.mdx

More details about this feature can be found in the [configuration page](/platforms/javascript/session-replay/configuration/#network-details).

> ## Custom scrubbing

Reviewer: ```suggestion
## Custom Scrubbing
```

> As of SDK version 7.53.0, `beforeAddRecordingEvent` was added that allows you to scrub and/or filter out recording events before they leave the browser. These events include console logs and network request and response data.

Reviewer 2: I'm aiming for something that describes "scrub" a little more. It might be a word we use internally a lot, but other people don't hear it as often and need the hint.

```suggestion
As of SDK version 7.53.0, `beforeAddRecordingEvent` was added that allows you to modify, scrub (to remove PII) or ignore recording events before they leave the browser. These events include console logs and network request and response data.
```

> The `beforeAddRecordingEvent` has been added starting with SDK, version 7.53.0. It allows you to modify, scrub the recordings to remove PII, or ignore recording events before they leave the browser. These events include console logs, network requests, and response data.

Reviewer 2: ```suggestion
The `beforeAddRecordingEvent` has been added starting with SDK version 7.53.0. It allows you to modify, scrub the recordings to remove PII, or ignore recording events before they leave the browser. These events include console logs, network requests, and response data.
```

```javascript
new Sentry.Replay({
  beforeAddRecordingEvent: (event) => {
    // Filter out specific events
    if (event.data.tag === "foo") {
      return null;
    }

    // Remember to return an event if you want to keep it!
    return event;
  },
});
```

> Note that we have server-side scrubbing for this data that will certain patterns such as American social security numbers, credit cards, and private keys.

Reviewer 3: ```suggestion
Note that we have [server-side scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data that will certain patterns such as American social security numbers, credit cards, and private keys.
```

Reviewer 2: Building on the previous suggestion by adding "PII" in there:
```suggestion
Note that we have [server-side PII scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data that will certain patterns such as American social security numbers, credit cards, and private keys.
```

> Note that we have [server-side PII scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data. It looks for certain patterns such as American social security numbers, credit cards, and private keys.

Reviewer 4: ```suggestion
Note that we also have [server-side PII scrubbing](/product/data-management-settings/scrubbing/server-side-scrubbing/) for this data. It looks for certain patterns such as American social security numbers, credit cards, and private keys.
```

not sure, something like this?
